JumpscaleLibs/sal/zosv2/billing.py

```diff
@@ -0,0 +1,60 @@
+from .resource import ResourceParser
+
+
+class Billing:
+    def __init__(self, explorer):
+        self._explorer = explorer
+
+    def reservation_resources(self, reservation):
+        """
+        compute how much resource units is reserved in the reservation
+
+        :param reservation: reservation object
+        :type reservation: tfgrid.workloads.reservation.1
+        :return: list of ResourceUnitsNode object
+        :rtype: list
+        """
+        rp = ResourceParser(self._explorer, reservation)
+        return rp.calculate_used_resources()
+
+    def reservation_resources_cost(self, reservation):
+        """
+        compute how much resource units is reserved in the reservation
+
+        :param reservation: reservation object
+        :type reservation: tfgrid.workloads.reservation.1
+        :return: list of ResourceUnitsNode object with costs filled in
+        :rtype: list
+        """
+        rp = ResourceParser(self._explorer, reservation)
+        return rp.calculate_used_resources_cost()
+
+    def payout_farmers(self, tfchain_wallet, resource_units_per_node, reservation):
```

Comment:
the `resource_units_per_node` parameter is not really ergonomic for a user to pass here imo. I think we can just leave it out, since the reservation itself is enough to calculate this. Removing this parameter, the second line of this function can just b e`resource_units_per_node = rp.calculate_used_resources_cost()`.

Reply:
indeed better UX without it.